src/pages/api/productDateInformation.ts

```diff
@@ -0,0 +1,169 @@
+            return 'start-date';
+        case 'startDateYear':
+            return 'start-date';
+        case 'endDateDay':
+            return 'end-date';
+        case 'endDateMonth':
+            return 'end-date';
+        case 'endDateYear':
+            return 'end-date';
+        case 'startDate':
+            return 'start-date';
+        case 'endDate':
+            return 'end-date';
+        default:
+            throw new Error(`Could not match the following error with an expected input. Error path: ${errorPath}.`);
+    }
+};
+
+const isDatesFieldEmpty = (day: string, month: string, year: string): boolean =>
+    day === '' && month === '' && year === '';
+
+export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
+    try {
+        let errors: ErrorInfo[] = [];
+
+        const {
+            startDateDay,
+            startDateMonth,
+            startDateYear,
+            endDateDay,
+            endDateMonth,
+            endDateYear,
+            productDates,
+        } = req.body;
+
+        const dateInput: ProductDateInformation = {
+            startDateDay,
+            startDateMonth,
+            startDateYear,
+            endDateDay,
+            endDateMonth,
+            endDateYear,
+        };
+
+        if (!productDates) {
+            errors.push({ errorMessage: 'Choose one of the options below', id: 'product-dates-required' });
+            updateSessionAttribute(req, PRODUCT_DATE_ATTRIBUTE, { errors, dates: dateInput });
+            redirectTo(res, '/productDateInformation');
+            return;
+        }
+
+        let startDate = moment();
+        let endDate = moment().add(100, 'y');
+
+        if (productDates === 'Yes') {
+            const isStartDateEmpty = isDatesFieldEmpty(startDateDay, startDateMonth, startDateYear);
+            const isEndDateEmpty = isDatesFieldEmpty(endDateDay, endDateMonth, endDateYear);
+
+            if (isStartDateEmpty && isEndDateEmpty) {
+                errors.push({ errorMessage: 'Enter a start or end date', id: 'start-date' });
+            }
+
+            if (!isStartDateEmpty) {
+                startDate = moment([startDateYear, startDateMonth - 1, startDateDay, '00', '01']);
+            }
+
+            if (!isEndDateEmpty) {
+                endDate = moment([endDateYear, endDateMonth - 1, endDateDay, '23', '59']);
+            }
+
+            if (!startDate.isValid() && !isStartDateEmpty) {
+                errors.push({ errorMessage: 'Start date must be a real date', id: 'start-date' });
+            }
+
+            if (!endDate.isValid() && !isEndDateEmpty) {
+                errors.push({ errorMessage: 'End date must be a real date', id: 'end-date' });
+            }
+
+            if (errors.length > 0) {
+                updateSessionAttribute(req, PRODUCT_DATE_ATTRIBUTE, { errors, dates: dateInput });
+                redirectTo(res, '/productDateInformation');
+                return;
+            }
```

Comment:
When we are attaching IDs to the errors in this API, we need to make sure they map to the correct element on the page to satisfy DAC requirements. Should we review these IDs and make sure they map correctly?

Reply:
yeah this will happen in the testing and seem to be working but worth double checking